nistats/second_level_model.py

```diff
@@ -0,0 +1,381 @@
+
+    verbose : integer, optional
+        Indicate the level of verbosity. By default, nothing is printed.
+        If 0 prints nothing. If 1 prints final computation time.
+        If 2 prints masker computation details.
+
+    n_jobs : integer, optional
+        The number of CPUs to use to do the computation. -1 means
+        'all CPUs', -2 'all CPUs but one', and so on.
+
+    """
+    def __init__(self, mask=None, smoothing_fwhm=None,
+                 memory=None, memory_level=1, verbose=0,
+                 n_jobs=1, minimize_memory=True):
+        self.mask = mask
+        self.smoothing_fwhm = smoothing_fwhm
+        self.memory = memory
+        self.memory_level = memory_level
+        self.verbose = verbose
+        self.n_jobs = n_jobs
+        self.minimize_memory = minimize_memory
+        self.labels_ = None
+        self.results_ = None
+
+    def fit(self, second_level_input, first_level_conditions=None,
+            confounds=None, design_matrix=None):
+        """ Fit the second-level GLM
+
+        1. create design matrix
+        2. do a masker job: fMRI_data -> Y
+        3. fit regression to (Y, X)
+
+        Parameters
+        ----------
+        second_level_input: list of `FirstLevelModel` objects or pandas
+                            DataFrame or list of Niimg-like objects.
+            If list of `FirstLevelModel` objects, then first_level_conditions
+            must be provided. If a pandas DataFrame, then must contain columns
+            subject_id, map_name, effects_map_path. If list of Niimg-like
+            objects then this is taken literally as Y for the model fit
+            and design_matrix must be provided.
+
+        first_level_conditions: list of (str, (str or array)) pairs
+                                or list of (str, str) pairs or None
+            If second_level_input is a list of `FirstLevelModel` objects then
+            it is mandatory to provide a list with contrast names as first item
+            (employed as column names in the design matrix) and contrast
+            definitions as second item. The contrast definitions are passed
+            to the compute_contrast method of `FirstLevelModel`. The contrast
+            definitions can be a str or array. Check the compute_contrast
+            documentation of `FirstLevelModel` for more details on the
+            contrast definitions.
+
+            If second_level_input is a pandas DataFrame then a list with
+            column names as first item (employed in the design matrix) and their
+            corresponding map name as second item, where the map names correspond
+            to those in the second_level_input map_name column.
+            If first_level_conditions is set to None then all maps are included
+            and the map_name is used as the column names in the design matrix.
+
+            If second_level_input is a list of Niimg-like objects then this
+            argument is ignored.
+
```

Comment:
This docstring is also hard to read, because it describes a complex logic. This is where a series of examples for model specification will be useful